### ISSUE TITLE
Performance/RedundantBlockCall errors out

### ISSUE THREAD

**Issue by stamhankar999** [bbatsov/rubocop#2730]
diag output: 0.36.0 (using Parser 2.3.0.2, running on ruby 1.9.3 x86_64-darwin15.0.0)

The error occurs when analyzing ruby-driver/lib/cassandra/types.rb. Clone https://github.com/datastax/ruby-driver.git and run rubocop in the lib directory.

**Comment by alexdowad** [comment]
Please post the stack trace here.

**Comment by stamhankar999** [comment]
here you go:
```
Scanning /Users/sandeeptamhankar/ruby-driver/lib/cassandra/types.rb
An error occurred while Performance/RedundantBlockCall cop was inspecting /Users/sandeeptamhankar/ruby-driver/lib/cassandra/types.rb.
undefined method `each_node' for nil:NilClass
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/performance/redundant_block_call.rb:36:in `blockarg_calls'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/performance/redundant_block_call.rb:40:in `block in on_def'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/performance/redundant_block_call.rb:30:in `blockarg_def'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/performance/redundant_block_call.rb:39:in `on_def'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:49:in `block (2 levels) in on_def'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:88:in `with_cop_error_handling'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:48:in `block in on_def'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:46:in `each'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:46:in `on_def'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:276:in `block in process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `map'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/parser-2.3.0.2/lib/parser/ast/processor.rb:9:in `process_regular_node'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:53:in `on_begin'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:276:in `block in process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `map'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/parser-2.3.0.2/lib/parser/ast/processor.rb:9:in `process_regular_node'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:53:in `on_class'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:276:in `block in process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `map'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/parser-2.3.0.2/lib/parser/ast/processor.rb:9:in `process_regular_node'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:53:in `on_begin'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:276:in `block in process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `map'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:275:in `process_all'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/parser-2.3.0.2/lib/parser/ast/processor.rb:9:in `process_regular_node'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:53:in `on_module'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/ast-2.2.0/lib/ast/processor/mixin.rb:259:in `process'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/commissioner.rb:54:in `investigate'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cop/team.rb:37:in `inspect_file'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:194:in `inspect_file'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:164:in `block in do_inspection_loop'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:158:in `loop'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:158:in `do_inspection_loop'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:87:in `process_file'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:59:in `block in inspect_files'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:57:in `each'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:57:in `inspect_files'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/runner.rb:35:in `run'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/lib/rubocop/cli.rb:30:in `run'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/bin/rubocop:14:in `block in <top (required)>'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/1.9.1/benchmark.rb:295:in `realtime'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/lib/ruby/gems/1.9.1/gems/rubocop-0.36.0/bin/rubocop:13:in `<top (required)>'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/bin/rubocop:23:in `load'
/Users/sandeeptamhankar/.rbenv/versions/1.9.3-p551/bin/rubocop:23:in `<main>'
```

**Comment by stamhankar999** [comment]
The code being analyzed is the Datstax Ruby driver for Cassandra. You can clone the repo and play with it: https://github.com/datastax/ruby-driver.git

I ran rubocop in the lib directory: `rubocop -d cassandra/types.rb`

**Comment by stamhankar999** [comment]
any updates on this?

**Comment by alexdowad** [comment]
Hmm. No error is produced on `master`. Looks like this may have already been fixed?

**Comment by jonas054** [comment]
It was fixed by #2651, which is included in v0.37.0.
